.circleci/config.yml

```diff
@@ -100,18 +100,19 @@ commands:
           name: Update git submodules
           command: git submodule update --init --recursive
 
-  halt_job_noncode_changes:
-    description: "Halt job and succeed early if no code changes on non-master branch"
+  halt_if_code_unchanged:
+    description: "Halt job if no code changed"
     steps:
       - run:
           command: |
-            if [ ${CIRCLE_BRANCH} != "" ] &&
-              [ ${CIRCLE_BRANCH} != "master" ] &&
-              ! git diff --name-only $(git merge-base origin/master ${CIRCLE_BRANCH}) | grep -qvE '(.md$)|(.pdf$)'; then
-                echo "No code changes on non-master branch. halting job."
-                circleci step halt
+            if ! git diff --name-only $(git merge-base origin/master ${CIRCLE_BRANCH}) | grep -qvFf .circleci/e2e-job-ignore-patterns.txt; then
+              echo "Application code have not changed. halting job."
+              circleci step halt
+            else
+              git diff --name-only $(git merge-base origin/master ${CIRCLE_BRANCH}) | grep -vFf .circleci/e2e-job-ignore-patterns.txt
+              echo "job continuing."
             fi
```

Comment:
Checking for PR branch `if [ ${CIRCLE_BRANCH} != "" ] && [ ${CIRCLE_BRANCH} != "master" ]` is now replaced by `pr-skip-ci.sh`.

The `grep` regex patterns is in `e2e-job-ignore-patterns.txt`.

<img width="1166" alt="Screen Shot 2020-06-23 at 7 46 14 PM" src="https://user-images.githubusercontent.com/35533885/85478453-45d62d80-b58a-11ea-9e3c-9975d07d8b8b.png">

Reply:
Very nice. I like how you didn't have to update a docker image to add your script. I was worried it would be worse.

Reply:
So, the `else` part in shell script is just to show you that non-matching files are the reason job is continuing. I can remove it if you like.